### DISCUSSION TITLE
index

### DISCUSSION THREAD

**Discussion by giscus** [mathieucarbou/YaSolR#1] [Website comments]
# index

Home

https://yasolr.carbou.me/

**Comment by mathieucarbou** [comment]
N'hésitez pas à poster vos commentaires et remarques sur les pages du site Web.

**Comment by RemixQ** [comment]
Hello.
First of all thank you for developing this awesome project that I follow for a long time now.
Do you plan to support modbus tcp as a grid power measure?

**Reply by mathieucarbou** [reply]
No because modbus is a protocol but there are different hardware using that and exposing different kind of data on different registers. So there is no generic way to get a grid power measurement over modbus. Also, it would require a ttl-rs485 converter.

YaSolR supports already many power measurement mechanisms, local or remote

**Reply by RemixQ** [reply]
Thank you for your reply.

Most inverters implement SunSpec modbus spec. Not sure to catch the point about ttl/rs485 converter in case of modbus tcp usage.

However yes, let's use other options.

**Reply by mathieucarbou** [reply]
What yasolr needs is not the inverter in/out: this is the grid power in/out.
Are you saying that because you are off-grid ?

**Reply by RemixQ** [reply]
No, but the inverter (Fronius in my case) exposes values of its smart meter through modbus as well (with an unknown frequency however, I plan to perform some tests).

**Reply by mathieucarbou** [reply]
you also know the register addresses, size, and their byte ordering ?

**Reply by RemixQ** [reply]
Not yet, I'll check in coming days.

**Reply by RemixQ** [reply]
So long story short, I gave up int+SF model type as it generates wrong values (due to some inconsistent Scale Factor values) and switch to float model type.

Here is an extract of my HA config:
```yaml
    - name: reading_ac_current_main_meter_float
      slave: 240
      scan_interval: 5
      address: 40071
      data_type: float32
      unit_of_measurement: A
      device_class: current
```

So register address 40071, type float32, big-endian order following Sunspec documentation.

The minimum scan_interval is 1 on HA but anyway the smart meter seems to send its value to the inverter each 5s only (which is a bit slow for a router).

**Reply by mathieucarbou** [reply]
And did you try to look at some integrations like this one: https://github.com/CJNE/ha-sunspec, or look at how they query the modbus device in order to replicate with ESPHome ?

Also, I don't know if it can help, but PVBrain support multiple inverters with ESPHome modbus config examples (https://github.com/SeByDocKy/pvbrain2/tree/main/pvbrain2/inverter). They do not have Sunspec examples, but just in case looking at the fields they use could help...

**Comment by luxxtek** [comment]
Hello, 
very interesting article, thank you for the effort & time.
I'm planning on building a solar router but still investigating the best configuration.
In this article I read 
** "If a bypass relay is installed, the dimmer will be used instead and will be set to 0-100% to simulate the relay."**
Is this correct or should it be ** "if no bypass relay is installed" **?

**Reply by mathieucarbou** [reply]
yes, thanks I will fix ;-)

**Comment by luc-c** [comment]
Bonjour Mathieu,
Bravo pour ce travail, on ressent bien expérience professionnelle dans toute l’implémentation qui se distingue des autres projets.
J'ai deux questions:
Dans les artefacts de compilation, on retrouve le module "fils" APPER sur wemos. Je n'ai pas trouvé d’élément y faisant reference. Est ce possible d'en savoir plus?
J'ai apprécié les cartes APPER avec le TTGO display pour son coté prêt a brancher. Ayant vu le build pour APPER wemos, est ce qu'il y a eu des builds pour TTGO? J'ai commencé de mon coté un portage pour utiliser l’écran mais je me suis confronté a des soucis sur les pins utilisables ce qui me fait douter de la viabilité du projet ... est ce quelque chose qui a été regardé? 

Merci

**Reply by mathieucarbou** [reply]
Bonjour,

Merci du compliment et bonne année !

Cette entrée a été contribuée par un utilisateur qui utilise YaSolR sur la carte de dimmer de l'APPER, mais juste cette carte. Elle ne prend pas de d'écran mais juste un esp32 et le robotdyn, ou ssr random + zcd, est connecté à cette carte avec éventuellement une ds18.

J'ai la carte également donc si tu souhaites  te lancer avec je peux aider. Je n'ai pas vérifié sur les gpio sont pré-configurés: vu que la carte prend un esp32 wemos de mémoire, à mon avis les gpio sont prédéfinis et pourrais être proposés par défaut dans YaSolR.

**Comment by iblackwood** [comment]
Bonjour !

This project looks great ! I have some questions I could not find covered in your excellent manual and build documents.
I have installed firmware on (Wemos) Lolin S3 Mini. These only have 4MB flash, 2MB RAM.
What is the minimum storage and memory you need for YASOLR ? 
Does the memory needed increase as you configure options ?

The ESP32-S3 and some other variants have built in temperature sensor. Have you considered using this ? I get that an external sensor is better for "whole of box" temperature though.

With the two other temperature sensors - is this for two separate heaters or upper and lower of one heater ?

Final question - I already have a PWM to 0-10V board but not the DFRobot one that just needs 5V. Mine needs 12V for output. Can I use this ? 

Mille mercis !

**Reply by iblackwood** [reply]
One more question I forgot ! (Pardon)
I have power meter (Eastron SDM230) connected to Home Assistant via RS422 / Modbus. How can I access this in Yasolr ? I assume via MQQT but cannot figure out how to publish that on the HA server.
Merci encore !

**Reply by mathieucarbou** [reply]
> Bonjour !
> 
> This project looks great ! I have some questions I could not find covered in your excellent manual and build documents.
> I have installed firmware on (Wemos) Lolin S3 Mini. These only have 4MB flash, 2MB RAM.
> What is the minimum storage and memory you need for YASOLR ? 
> Does the memory needed increase as you configure options ?
> 
> The ESP32-S3 and some other variants have built in temperature sensor. Have you considered using this ? I get that an external sensor is better for "whole of box" temperature though.
> 
> With the two other temperature sensors - is this for two separate heaters or upper and lower of one heater ?
> 
> Final question - I already have a PWM to 0-10V board but not the DFRobot one that just needs 5V. Mine needs 12V for output. Can I use this ? 
> 
> Mille mercis !
> 

Hi,
To generally answer your questions: stick with only the supported hardware. If it is not listed the it is not supported.

**Reply by mathieucarbou** [reply]
> One more question I forgot ! (Pardon)
> I have power meter (Eastron SDM230) connected to Home Assistant via RS422 / Modbus. How can I access this in Yasolr ? I assume via MQQT but cannot figure out how to publish that on the HA server.
> Merci encore !

Yes you need to have mqtt setup on HA and use mqtt stream feature of HA:

https://www.home-assistant.io/integrations/mqtt_statestream/

Important to use include to only expose what is necessary otherwise all your ha sensors will be exposed.
